codec/codec-fury/src/main/java/com/alipay/sofa/rpc/codec/fury/FurySerializer.java

```diff
@@ -0,0 +1,326 @@
+    public Object decode(final AbstractByteBuf data, final Class clazz, final Map<String, String> context)
+        throws SofaRpcException {
+        Object result = null;
+        if (clazz == null) {
+            throw buildDeserializeError("class is null!");
+        } else if (data.readableBytes() <= 0) {
+            try {
+                result = clazz.newInstance();
+            } catch (InstantiationException | IllegalAccessException e) {
+                throw buildDeserializeError(e.getMessage());
+            }
+            return result;
+        } else {
+            result = fury.deserialize(data.array());
+        }
+        return result;
+    }
+
+    @Override
+    public void decode(final AbstractByteBuf data, final Object template, final Map<String, String> context)
+        throws SofaRpcException {
+        if (template == null) {
+            throw buildDeserializeError("template is null!");
+        } else if (template instanceof SofaRequest) {
+            decodeSofaRequest(data, (SofaRequest) template, context);
+        } else if (template instanceof SofaResponse) {
+            decodeSofaResponse(data, (SofaResponse) template, context);
+        } else {
+            throw buildDeserializeError("Only support decode from SofaRequest and SofaResponse template");
+        }
+
+    }
+
+    private void decodeSofaRequest(AbstractByteBuf data, SofaRequest sofaRequest, Map<String, String> head) {
+        if (head == null) {
+            throw buildDeserializeError("head is null!");
+        }
+        String targetService = head.remove(RemotingConstants.HEAD_TARGET_SERVICE);
+        if (targetService != null) {
+            sofaRequest.setTargetServiceUniqueName(targetService);
+            String interfaceName = ConfigUniqueNameGenerator.getInterfaceName(targetService);
+            sofaRequest.setInterfaceName(interfaceName);
+        } else {
+            throw buildDeserializeError("HEAD_TARGET_SERVICE is null");
+        }
+        String methodName = head.remove(RemotingConstants.HEAD_METHOD_NAME);
+        if (methodName != null) {
+            sofaRequest.setMethodName(methodName);
+        } else {
+            throw buildDeserializeError("HEAD_METHOD_NAME is null");
+        }
+        String targetApp = head.remove(RemotingConstants.HEAD_TARGET_APP);
+        if (targetApp != null) {
+            sofaRequest.setTargetAppName(targetApp);
+        }
+
+        // parse tracer and baggage
+        parseRequestHeader(RemotingConstants.RPC_TRACE_NAME, head, sofaRequest);
+        if (RpcInvokeContext.isBaggageEnable()) {
+            parseRequestHeader(RemotingConstants.RPC_REQUEST_BAGGAGE, head, sofaRequest);
+        }
+        for (Map.Entry<String, String> entry : head.entrySet()) {
+            sofaRequest.addRequestProp(entry.getKey(), entry.getValue());
+        }
+
+        // according interface and method name to find parameter types
+        Class[] requestClass = furyHelper.getReqClass(targetService,
```

Comment:
just use sofaRequest.getMethodArgs() to get arg names, and ClasstypeUtils to get class is fine